src/cli-client/internals/client/export_test.go

```diff
@@ -0,0 +1,13 @@
+package client
+
+var (
+	GetWorkflowRunID                = getWorkflowRunID
+	GetWorkflowRunStatusFromResp    = getWorkflowRunStatusFromResp
+	GetWorkflowJobsProgressFromResp = getWorkflowJobsProgressFromResp
+)
+
+func SetWorkflowDispatchURL(url string) string {
+	ret := workflowDispatchURL
+	workflowDispatchURL = url
+	return ret
+}
```

Comment:
One nice pattern from Pebble (not sure if it's a general Go pattern) is that the returned value will be a callable, so you don't need to know how to restore it, [example from Pebble](https://github.com/canonical/pebble/blob/d822e9fd9c484a9c35e1b8f08b8d438ca2115225/internals/overlord/state/task.go#L413-L416) and [its usage](https://github.com/canonical/pebble/blob/d822e9fd9c484a9c35e1b8f08b8d438ca2115225/internals/overlord/state/task_test.go#L357-L358):

```suggestion
func SetWorkflowDispatchURL(url string) (restore func()) {
	orig := workflowDispatchURL
	workflowDispatchURL = url
	return func() { workflowDispatchURL = orig }
}
```

Reply:
Nice one. Fixed.